components/datepicker/modules/calendar.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <div class="d-datepicker__calendar">
+    {{ weekDays }}
+    <div
+      v-for="(week, indexWeek) in calendarDays"
+      :key="indexWeek"
+    >
+      {{ indexWeek }}
+      <button
+        v-for="(day, indexDays) in week.days"
+        :key="indexWeek + indexDays"
+        type="button"
+        :aria-label="day.text"
+        @click="selectDay(day)"
+        @keydown="$emit('select-date', day.value)"
+      >
+        {{ day.text }}
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import { getDayNames } from '@/components/datepicker/utils.js';
+import { WEEK_START } from '@/components/datepicker/datepicker_constants.js';
+
+export default {
+  name: 'DtDatepickerCalendar',
+
+  props: {
+    calendarDays: {
+      type: Array,
+      required: true,
+    },
+
+    locale: {
+      type: String,
+      required: true,
+    },
+  },
+
+  emits: [
+    /**
+     * Event fired when a date is selected
+     *
+     * @event select-date
+     * @type {Date}
+     */
+    'select-date',
+  ],
+
+  data () {
+    return {
+      thisCalendarDays: this.calendarDays,
+    };
+  },
+
+  computed: {
+    weekDays () {
+      return getDayNames(this.locale, WEEK_START);
+    },
+  },
+
+  watch: {},
+
+  methods: {
+    selectDay (day) {
+      if (!day.current) { return; }
```

Comment:
This is because `current` means the current month.
We are placing all days in the `calendar` components.

<img width="329" alt="Screenshot 2023-06-13 at 12 02 37" src="https://github.com/dialpad/dialtone-vue/assets/89984179/cabb200f-ee91-483e-ba58-fae08eaabaa5">


Since we are not handle the automatic month change if you click those days, then it should not be clicked.
These days will be disabled with a gray color (I'm assuming this because the design did not handle this and we didnt groom this behaviour). Something like this:

<img width="253" alt="image" src="https://github.com/dialpad/dialtone-vue/assets/89984179/8ff6cc02-eb8a-4689-b2f0-827456f2952d">

Reply:
I updated `current` to `currentMonth` to be more descriptive